docs/02-customize-a-pizza.md

```diff
@@ -282,11 +290,14 @@ void CancelConfigurePizzaDialog()
 {
     configuringPizza = null;
     showingConfigureDialog = false;
-    StateHasChanged();
 }
 ```
 
-The `StateHasChanged` method signals to the runtime that the component's state has changed and it needs to be rendered. Components are rendered automatically by the runtime when its parameters change or when a UI event is fired on that component. In this case the event triggering the state change came from a different component, so `StateHasChanged` needs to be called manually.
+Now, what happens when you click the dialog cancel button is that `Index.CancelConfigurePizzaDialog` will execute, and then the `Index` component will render itself. Since `showingConfigureDialog` is now `false` the dialog will not be displayed. 
+
+Normally what happens when you trigger an event (like clicking the cancel button) is that the component that defined the event handler delegate will rerender. You could define events using any delegate type like `Action` or `Func<string, Task>`. Sometimes you want to use an event handler delegate that doesn't belong to a component - if you used a normal delegate type to define the event then nothing will be rendered or updated. 
+
+`EventCallback` is a special type that is known to the compiler that resolves some of these issues. It tells the compiler to find the *best* component component to attach the event handler to so that the UI will update. `EventCallback` has a few more tricks up its sleeve, but for now just remember that using `EventCallback` makes your component smart about dispatching events to the right place.
```

Comment:
I wanted to use this section to demonstrate the concept behind `EventCallback` but I can't because all of these cases use bound-delegates and so they work without `EventCallback`. 

I'm doing section 4 next and I'm going to see if `AppState` starts to make more sense now that we have `EventCallback`. If so that should provide an opportunity to explain this.

It feels good going through and removing `StateHasChanged()` from the examples 😁

Reply:
> It feels good going through and removing StateHasChanged() from the examples

👍 

> It tells the compiler to find the *best* component component to attach the event handler to so that the UI will update.

That feels a bit ambiguous. Maybe something like:

```
It captures information about which component contains the callback's logic, and causes *that* to be the component that re-renders when it executes.
```